app/views/layouts/menus/_ta_sub_menu.html.erb

```diff
@@ -1,27 +1,72 @@
+<% is_viewing_a_peer_review_result = !@result.nil? && @result.is_a_review? %>
+
 <ul class='sub'>
   <li id='dropdown'>
-      <span class='dropdown'>
-        <% if @assignment.nil? && @grade_entry_form.nil? %>
-          <%= t('menu.choose') %>
-        <% else %>
-          <%= @assignment.is_peer_review? ?
-                @assignment.parent_assignment.short_identifier + ' ' + PeerReview.model_name.human :
-                @assignment.short_identifier %>
+    <div class='dropdown'>
+      <%# Display an appropriate message in the dropdown menu %>
+      <% if assessment.nil? %>
+        <%= t('menu.choose') %>
+      <% else %>
+        <% title = assessment.short_identifier %>
+        <% if assessment.is_a?(Assignment) && assessment.is_peer_review? %>
+          <% title = "#{assessment.parent_assignment.short_identifier} #{PeerReview.model_name.human}" %>
         <% end %>
-        <ul>
-           <%= render partial: 'shared/assignments_dropdown_menu' %>
-        </ul>
-      </span>
+        <% if assessment.is_hidden %>
+          <% title = t('assignments.hidden', assignment_text: title) %>
+        <% end %>
+        <%= title %>
+      <% end %>
+      <ul>
+        <%= render partial: 'shared/assignments_dropdown_menu' %>
+      </ul>
+    </div>
   </li>
-  <% if controller.controller_name == 'submissions' ||
-    controller.controller_name == 'results' ||
-    controller.action_name == 'summary' %>
-    <li <% if controller.controller_name == 'submissions' ||
-      controller.controller_name == 'results' %>
-        class='active'
-        <% end %>>
-      <%= link_to Submission.model_name.human.pluralize,
-                  browse_assignment_submissions_path(@assignment || @grouping.assignment) %>
+
+  <% if assessment.is_a?(GradeEntryForm) && !assessment.new_record? %>
```

Comment:
A lot of the code below is copied from the admin view. Instead of copying code. Please find a way to combine the two views or factor out the shared code.